homeassistant/helpers/template.py

```diff
@@ -1956,6 +1956,43 @@ def is_number(value):
     return True
 
 
+def is_list(value):
+    """Return whether a value is a list."""
+    return isinstance(value, list)
+
+
+def is_set(value):
+    """Return whether a value is a set."""
+    return isinstance(value, set)
+
+
+def is_tuple(value):
+    """Return whether a value is a tuple."""
+    return isinstance(value, tuple)
+
+
+def _set(value):
+    """Convert value to set."""
+    return set(value)
+
+
+def _tuple(value):
+    """Convert value to tuple."""
+    return tuple(value)
+
+
+def is_datetime(value):
+    """Return whether a value is a datetime."""
+    return isinstance(value, datetime)
+
+
+def is_non_string_iterable(value):
+    """Return whether a value is an iterable that's not a string."""
+    return isinstance(value, Iterable) and not isinstance(
+        value, (str, bytes, bytearray)
+    )
```

Comment:
To continue the discussion on `non_string_iterable`...
I also don't like the name, and I think we should not create `non_*` tests, when jinja has bultin `not` support.
To keep consistency over all tests I would suggest something like:

```suggestion
def is_stringlike(value):
    """Return whether a value is str, bytes, bytearray."""
    return isinstance(
        value, (str, bytes, bytearray)
    )
```

So users can write

```jinja
{% set value = "test" %}
{{ value is iterable and value is not stringlike }}
```

What do you think?

Reply:
works for me, updated